exporters/zipkin/src/test/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporterEndToEndHttpTest.java

```diff
@@ -0,0 +1,195 @@
+  private static final Map<String, AttributeValue> attributes = Collections.emptyMap();
+  private static final List<SpanData.TimedEvent> annotations =
+      ImmutableList.of(
+          SpanData.TimedEvent.create(
+              1505855799_433901068L, "RECEIVED", Collections.<String, AttributeValue>emptyMap()),
+          SpanData.TimedEvent.create(
+              1505855799_459486280L, "SENT", Collections.<String, AttributeValue>emptyMap()));
+
+  private static final String ENDPOINT_V1_SPANS = "/api/v1/spans";
+  private static final String ENDPOINT_V2_SPANS = "/api/v2/spans";
+  private static final String SERVICE_NAME = "myService";
+  private static final Endpoint localEndpoint =
+      ZipkinSpanExporter.produceLocalEndpoint(SERVICE_NAME);
+
+  @Rule public ZipkinRule zipkin = new ZipkinRule();
+
+  @Test
+  public void testExportWithDefaultEncoding() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setEndpoint(zipkin.httpUrl() + ENDPOINT_V2_SPANS)
+            .setServiceName(SERVICE_NAME)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportAsProtobuf() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setSender(
+                URLConnectionSender.newBuilder()
+                    .endpoint(zipkin.httpUrl() + ENDPOINT_V2_SPANS)
+                    .encoding(Encoding.PROTO3)
+                    .build())
+            .setServiceName(SERVICE_NAME)
+            .setEncoder(SpanBytesEncoder.PROTO3)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportAsThrift() {
+
+    @SuppressWarnings("deprecation")
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setSender(
+                URLConnectionSender.newBuilder()
+                    .endpoint(zipkin.httpUrl() + ENDPOINT_V1_SPANS)
+                    .encoding(Encoding.THRIFT)
+                    .build())
+            .setServiceName(SERVICE_NAME)
+            .setEncoder(SpanBytesEncoder.THRIFT)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportAsJsonV1() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setSender(
+                URLConnectionSender.newBuilder()
+                    .endpoint(zipkin.httpUrl() + ENDPOINT_V1_SPANS)
+                    .encoding(Encoding.JSON)
+                    .build())
+            .setServiceName(SERVICE_NAME)
+            .setEncoder(SpanBytesEncoder.JSON_V1)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportFailedAsWrongEncoderUsed() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
```

Comment:
```
        ZipkinExporterConfiguration.builder()
            .setSender(
                URLConnectionSender.newBuilder()
                    .endpoint(zipkin.httpUrl() + ENDPOINT_V2_SPANS)
                    .encoding(Encoding.JSON)
                    .build())
            .setServiceName(SERVICE_NAME)
            .setEncoder(SpanBytesEncoder.PROTO3)
            .build();
```
looking like it can be method with 4 parameters instead of duplicating it

Reply:
While tidying the test up I extracted this into its own method